src/git-directory-helper.ts

```diff
@@ -56,10 +56,24 @@ export async function prepareExistingDirectory(
         await git.branchDelete(false, branch)
       }
 
-      // Remove all refs/remotes/origin/* to avoid conflicts
-      branches = await git.branchList(true)
-      for (const branch of branches) {
-        await git.branchDelete(true, branch)
+      // Remove any conflicting refs/remotes/origin/*
```

Comment:
Detects two types of conflicts:

Example 1: Consider `ref` is `refs/heads/foo` and an existing ref `refs/remotes/origin/foo/bar` exists

Example 2: Consider `ref` is `refs/heads/foo/bar` and an existing ref `refs/remotes/origin/foo` exists

Reply:
todo: add examples to comments in the code